crs_application/src/task_managers/motion_planning_manager.cpp

```diff
@@ -53,37 +68,246 @@ MotionPlanningManager::~MotionPlanningManager()
 {
-  RCLCPP_WARN(node_->get_logger(),"%s not implemented yet",__PRETTY_FUNCTION__);
+  common::ActionResult res = checkPreReq();
+  if( !res )
+  {
+    return res;
+  }
+
+  // computing split distance
+  const double split_dist = config_->tool_speed * config_->media_change_time;
+
+  // splitting paths
+  std::vector<datatypes::ProcessToolpathData> toolpaths_processes;
+  double current_dist = 0.0;
+  Eigen::Vector3d p1, p2;
+  toolpaths_processes.resize(1);
+  for(std::size_t r_idx = 0 ; r_idx < input_->rasters.size(); r_idx++)
+  {
+    geometry_msgs::msg::PoseArray& raster = input_->rasters[r_idx];
+    int split_point_idx = -1;
+    for(std::size_t p_idx = 0; p_idx < raster.poses.size() - 1; p_idx++)
+    {
+      p1 = toEigen(raster.poses[p_idx].position);
+      p2 = toEigen(raster.poses[p_idx + 1].position);
+
+      double d = (p2 - p1).norm();
+      if(current_dist + d > split_dist)
+      {
+        RCLCPP_INFO(node_->get_logger(),"Found split at point %lu of raster %lu",p_idx, r_idx);
+        split_point_idx = p_idx;
+        current_dist = 0;
+        break;
+      }
+      current_dist += d;
+    }
+
+    if(split_point_idx >=0)
+    {
+      std::remove_reference <decltype(raster)>::type new_raster;
+      std::copy(raster.poses.begin(), raster.poses.begin() + split_point_idx,
+                std::back_inserter(new_raster.poses));
+      toolpaths_processes.back().rasters.push_back(new_raster);
+
+      // copying remaining part of raster into a new  process
+      toolpaths_processes.resize(toolpaths_processes.size() + 1);
+      new_raster.poses.clear();
+      std::copy(raster.poses.begin() + split_point_idx + 1, raster.poses.end(),
+                std::back_inserter(new_raster.poses));
+      toolpaths_processes.back().rasters.push_back(new_raster);
+    }
+    else
+    {
+      // no split so copy current raster into current process
+      toolpaths_processes.back().rasters.push_back(raster);
+    }
+  }
+  RCLCPP_INFO(node_->get_logger(),"Split original process into %lu", toolpaths_processes.size());
+  process_toolpaths_ = std::move(toolpaths_processes);
   return true;
 }
 
 common::ActionResult MotionPlanningManager::planProcessPaths()
 {
-  RCLCPP_WARN(node_->get_logger(),"%s not implemented yet",__PRETTY_FUNCTION__);
+  using namespace crs_msgs;
+
+  common::ActionResult res = checkPreReq();
+  if( !res )
+  {
+    return res;
+  }
+
+  // copying data into request
+  srv::PlanProcessMotions::Request::SharedPtr req = std::make_shared<srv::PlanProcessMotions::Request>();
+  for(datatypes::ProcessToolpathData& pd : process_toolpaths_)
+  {
+    msg::ToolProcessPath process_path;
+    for(const geometry_msgs::msg::PoseArray& raster : pd.rasters)
+    {
+      process_path.rasters.push_back(raster);
+    }
+    req->process_paths.push_back(process_path);
+  }
+
+  std::shared_future<srv::PlanProcessMotions::Response::SharedPtr> fut = process_motion_planning_client_->async_send_request(
```

Comment:
The request message is only being populated with process paths, the config settings approach dist, retreat dist, etc should also be added to the request message